src/lib/reasoners/satml_frontend.ml

```diff
@@ -1015,9 +1015,9 @@ module Make (Th : Theory.S) : Sat_solver_sig.S = struct
         unsat_rec env ~first_call:false
       with
       | Satml.Unsat lc -> raise (IUnsat (env, make_explanation lc))
+      | I_dont_know _ as e -> raise e
       | _ -> assert false
```

Comment:
Testing is a bit pointless because if testing *with* the `assert false` does not fail, it means no exception is raised, and so we will not see the difference, no?

Having the catch-all `assert false` here means that if the user hits `Ctrl-C` (`Sys.Break`), or if the timeout triggers within this specific piece of code (`Util.Timeout` -- it is inside the solver so I believe the timeout is active while we execute this code), we will die with an assertion failure instead, which is unexpected and prevents normal handling.

By the way, there are no uncatchable exceptions:
```ocaml
# try assert false with Assert_failure _ -> () ;;
- : unit = ()
```

Reply:
I know there is no uncatchable exceptions but catching an assert false is a big mistake :) 

You are right. It should actually fail :)